spring-cloud-gcp-bigquery/src/main/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -0,0 +1,279 @@
+	 * @param datasetNameExpression name of the BigQuery dataset
+	 */
+	public void setDatasetNameExpression(Expression datasetNameExpression) {
+		Assert.notNull(datasetNameExpression, "Dataset name expression can't be null.");
+		this.datasetNameExpr = datasetNameExpression;
+	}
+
+	/**
+	 * Sets the default BigQuery dataset name to use if it is omitted from message headers.
+	 * @param datasetName name of the BigQuery dataset
+	 */
+	public void setDatasetName(String datasetName) {
+		this.datasetNameExpr = new LiteralExpression(datasetName);
+	}
+
+	/**
+	 * Sets the SpEL expression to evaluate to determine the default table name if the
+	 * table name is omitted from message headers.
+	 * @param tableNameExpression name of the BigQuery dataset
+	 */
+	public void setTableNameExpression(Expression tableNameExpression) {
+		this.tableNameExpr = tableNameExpression;
+	}
+
+	/**
+	 * Sets the default BigQuery table name to use if it is omitted from message headers.
+	 * @param tableName name of the BigQuery dataset
+	 */
+	public void setTableName(String tableName) {
+		this.tableNameExpr = new LiteralExpression(tableName);
+	}
+
+	/**
+	 * Sets the default {@link FormatOptions} to use if it is omitted from message headers.
+	 * The {@link FormatOptions} describe the type/format of data files being loaded.
+	 * @param formatOptions the format of the data file being loaded
+	 */
+	public void setFormatOptions(FormatOptions formatOptions) {
+		this.formatOptions = formatOptions;
+	}
+
+	/**
+	 * Sets the {@link WriteDisposition} which specifies how data should be inserted into
+	 * BigQuery tables.
+	 * @param writeDisposition whether data should be appended or truncated to the BigQuery table
+	 */
+	public void setWriteDisposition(WriteDisposition writeDisposition) {
+		this.writeDisposition = writeDisposition;
+	}
+
+	/**
+	 * Sets whether BigQuery should attempt to autodetect the schema of the data when loading
+	 * data into an empty table for the first time.
+	 * @param autoDetectSchema whether data schema should be autodetected from the structure of
+	 * 		the data
+	 */
+	public void setAutoDetectSchema(boolean autoDetectSchema) {
+		this.autoDetectSchema = autoDetectSchema;
+	}
+
+	/**
+	 * Sets the {@link Duration} to wait for a file to be loaded into BigQuery before timing out
+	 * when waiting synchronously.
+	 * @param timeout the {@link Duration} timeout to wait for a file to load
+	 */
+	public void setTimeout(Duration timeout) {
+		this.timeout = timeout;
+	}
+
+	/**
+	 * A {@code boolean} indicating if the {@link BigQueryFileMessageHandler} should synchronously
+	 * wait for each file to be successfully loaded to BigQuery.
+	 * @param sync whether {@link BigQueryFileMessageHandler} should wait synchronously for jobs to
+	 * 		complete
+	 */
+	public void setSync(boolean sync) {
+		this.sync = sync;
+	}
+
+	@Override
+	protected ListenableFuture handleRequestMessage(Message<?> message) {
```

Comment:
You should consider do not return `ListenableFuture` in the `sync` mode.

So, just `return result.get(this.timeout.getSeconds(), TimeUnit.SECONDS);` should be enough for that case.

Also would be better do not use raw types as long as possible.